### PR TITLE
hotfix: added validation on infra config update

### PR DESCRIPTION
**Pull request by mirarifhasan**
### Description

In this PR added some validation when updating any variable in InfraConfig.

Some validations are:
1. SSO secrets can not be updated with the `null` or `""` value.
2. Any SSO can not be enabled if its secret/client_id is not configured beforehand.

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### Additional Information

Nil

### CODE REVIEW

File: packages/hoppscotch-backend/src/infra-config/infra-config.service.ts [hunk 1]
```diff
@@ -182,6 +184,51 @@ export class InfraConfigService implements OnModuleInit {
...
 
+  /**
+   * Check if the service is configured or not
+   * @param service Service can be Auth Provider, Mailer, Audit Log etc.
+   * @returns Either true or false
+   */
+  async isServiceConfigured(service: AuthProvider) {
+    switch (service) {
+      case AuthProvider.GOOGLE:
+        if (
```
Comment (ankitsridhar16) [inline packages/hoppscotch-backend/src/infra-config/infra-config.service.ts:195]: can further branching be written in a more abstract way ?

Comment (mirarifhasan) [reply packages/hoppscotch-backend/src/infra-config/infra-config.service.ts:195]: Can you give any suggestions?

Comment (ankitsridhar16) [reply packages/hoppscotch-backend/src/infra-config/infra-config.service.ts:195]: A helper method that could abstract away the common logic, where we could pass values as arguments maybe ?

**Review by balub — APPROVED** [review]

File: packages/hoppscotch-backend/src/infra-config/infra-config.service.ts (hunk 1 above)
Comment (mirarifhasan) [reply packages/hoppscotch-backend/src/infra-config/infra-config.service.ts:195]: How this looks, @ankitsridhar16?
```ts
isServiceConfigured(service: AuthProvider | 'AUDIT_LOG') {
  const configService = this.configService;

  switch (service) {
    case AuthProvider.GOOGLE:
      return configService.get<string>('INFRA.GOOGLE_CLIENT_ID') &&
             configService.get<string>('INFRA.GOOGLE_CLIENT_SECRET');

    case AuthProvider.GITHUB:
      return configService.get<string>('INFRA.GITHUB_CLIENT_ID') &&
             configService.get<string>('INFRA.GITHUB_CLIENT_SECRET');

    case AuthProvider.MICROSOFT:
      return configService.get<string>('INFRA.MICROSOFT_CLIENT_ID') &&
             configService.get<string>('INFRA.MICROSOFT_CLIENT_SECRET');

    case AuthProvider.EMAIL:
      return configService.get<string>('INFRA.MAILER_SMTP_URL') &&
             configService.get<string>('INFRA.MAILER_ADDRESS_FROM');

    case AuthProvider.SAML:
      return configService.get<string>('INFRA.SAML_ISSUER') &&
             configService.get<string>('INFRA.SAML_CERT') &&
             configService.get<string>('INFRA.SAML_ENTRY_POINT');

    case 'AUDIT_LOG':
      return !!configService.get<string>('INFRA.CLICKHOUSE_HOST');

    default:
      return false;
  }
}
```

Comment (ankitsridhar16) [reply packages/hoppscotch-backend/src/infra-config/infra-config.service.ts:195]: 👍

**Review by ankitsridhar16 — APPROVED** [review]

**Pull request merged**